src/Microsoft.SqlTools.ServiceLayer/AzureBlob/BlobService.cs

```diff
@@ -60,16 +60,24 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                 await requestContext.SendError(SR.NotSupportedCloudCreateSas);
                 return;
             }
-            using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "AzureBlob"))
+            try
             {
-                // Connection gets disconnected when backup is done
-                ServerConnection serverConnection = new ServerConnection(sqlConn);
-                Server sqlServer = new Server(serverConnection);
+                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "AzureBlob"))
+                {
+                    // Connection gets disconnected when backup is done
+                    ServerConnection serverConnection = new ServerConnection(sqlConn);
+                    Server sqlServer = new Server(serverConnection);
 
-                SharedAccessSignatureCreator sharedAccessSignatureCreator = new SharedAccessSignatureCreator(sqlServer);
-                string sharedAccessSignature = sharedAccessSignatureCreator.CreateSqlSASCredential(optionsParams.StorageAccountName, optionsParams.BlobContainerKey, optionsParams.BlobContainerUri, optionsParams.ExpirationDate);
-                response.SharedAccessSignature = sharedAccessSignature;
-                await requestContext.SendResult(response);
+                    SharedAccessSignatureCreator sharedAccessSignatureCreator = new SharedAccessSignatureCreator(sqlServer);
+                    string sharedAccessSignature = sharedAccessSignatureCreator.CreateSqlSASCredential(optionsParams.StorageAccountName, optionsParams.BlobContainerKey, optionsParams.BlobContainerUri, optionsParams.ExpirationDate);
+                    response.SharedAccessSignature = sharedAccessSignature;
+                    await requestContext.SendResult(response);
+                }
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex);
```

Comment:
Don't need to do this - we already catch any uncaught exceptions for requests and send an error back in the MessageDispatcher

https://github.com/microsoft/sqltoolsservice/blob/38f33fa211086d0def50b016dd6940b942ac9255/src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs#L162

Reply:
That's good to know!

Reply:
It now makes me think, should we always let this be handled by dispatcher then? 🤔

Reply:
That's what I prefer for messages that are set up to expect errors to be thrown. But some messages have a "success" property that's used to determine if failures happen (like [agent](https://github.com/microsoft/sqltoolsservice/pull/2190/files#diff-81a70bd053fe2c2d1385f7bad2a4213e86b421ea8d13a35a151b6c8d77423e8cR180) and so we should follow the pattern there for those that use that unless you verify that they can also handle errors being thrown in a clean manner.